grails-app/controllers/org/transmartproject/rest/ExportController.groovy

```diff
@@ -29,29 +29,27 @@ class ExportController {
      *  Returns datatypes and patient number of given concepts.
      *
      */
-    def datatypes(){
-        def jsonSlurper = new JsonSlurper()
+    def datatypes() throws NoSuchResourceException {
         if (!(params.containsKey('concepts'))){
-            throw new NoSuchElementException(
-                    "No parameter named concepts."
-            )
+            throw new NoSuchResourceException("No parameter named concepts was given.")
+        }
+        if (params.get('concepts') == "") {
+            throw new InvalidArgumentsException("Parameter concepts has no value.")
         }
-        def test = params.get('concepts').decodeURL()
+        def jsonSlurper = new JsonSlurper()
+        def conceptParameters = params.get('concepts').decodeURL()
         try {
-            def concept_arguments = jsonSlurper.parseText(test)
-            if (concept_arguments==null){
-                throw new NullArgumentException(
-                        "Parameter concepts has no value."
-                )
-            }
-            List datatypes = []
-            concept_arguments.each { it ->
+            def conceptArguments = jsonSlurper.parseText(conceptParameters)
+            List dataTypes = []
+            int cohortNumber = 1
+            conceptArguments.each { it ->
                 List conceptKeysList = it.conceptKeys
-                datatypes += restExportService.getDataTypes(conceptKeysList)
+                dataTypes = restExportService.getDataTypes(conceptKeysList, dataTypes, cohortNumber)
+                cohortNumber += 1
             }
-            respond(restExportService.formatDataTypes(datatypes))
+            respond(dataTypes)
```

Comment:
stylistic remark: in this call you use brackets - in the other one you use groovy bracketless call (index call). This is inconsistent, would look better to use just one way everywhere. (but this is not a big deal)